components/brave_wallet_ui/components/desktop/views/crypto/index.tsx

```diff
@@ -135,8 +137,8 @@ const CryptoView = (props: Props) => {
         onClickSettings={onClickSettings}
         onClickMore={onClickShowMore}
       />
-      {(defaultWallet !== BraveWallet.DefaultWallet.BraveWallet &&
-        (defaultWallet !== BraveWallet.DefaultWallet.BraveWalletPreferExtension || (defaultWallet === BraveWallet.DefaultWallet.BraveWalletPreferExtension && isMetaMaskInstalled))) &&
+      {((defaultEthereumWallet !== BraveWallet.DefaultWallet.BraveWallet || defaultSolanaWallet !== BraveWallet.DefaultWallet.BraveWallet) &&
+        (defaultEthereumWallet !== BraveWallet.DefaultWallet.BraveWalletPreferExtension || defaultSolanaWallet !== BraveWallet.DefaultWallet.BraveWalletPreferExtension || (defaultEthereumWallet === BraveWallet.DefaultWallet.BraveWalletPreferExtension && isMetaMaskInstalled))) &&
```

Comment:
Since this is a complex condition, could we make this its own `memo`?

Something like
```
  const showBanner = React.useMemo((): boolean => {
    return (
      (defaultEthereumWallet !== BraveWallet.DefaultWallet.BraveWallet ||
        defaultSolanaWallet !== BraveWallet.DefaultWallet.BraveWallet) &&
      (defaultEthereumWallet !== BraveWallet.DefaultWallet.BraveWalletPreferExtension ||
        defaultSolanaWallet !== BraveWallet.DefaultWallet.BraveWalletPreferExtension ||
        (defaultEthereumWallet === BraveWallet.DefaultWallet.BraveWalletPreferExtension &&
          isMetaMaskInstalled))) &&
      showDefaultWalletBanner
  }, [defaultEthereumWallet, defaultSolanaWallet, isMetaMaskInstalled, showDefaultWalletBanner])
```

and then all we need for the conditional is 
```
      {showBanner &&
        <WalletBanner
        ........
```